### ISSUE TITLE
Cannot  combine all scorefiles

### ISSUE THREAD

**Issue by mfasold** [PGScatalog/pgscatalog_utils#85]
Following [this discussion](https://github.com/PGScatalog/pgsc_calc/discussions/247), I am trying to create one big scorefile which combines *all* scorefiles from the catalog.

Hence, I first downloaded all individual scorefiles using

`for pgsid in $(cat pgs_scores_list.txt); do echo $pgsid; pgscatalog-download -i $pgsid -o . -b GRCh38`

I then try to combine them using

`pgscatalog-combine -s PGS*.txt.gz -t GRCh38 -o combined.txt`

This leads to the following error:

```
...
  File "XXX/venv/lib/python3.11/site-packages/pgscatalog/core/lib/genomebuild.py", line 51, in from_string
    raise ValueError(f"Can't match {build=}")
ValueError: Can't match build='NCBI35'
```
Any ideas on how to proceed?

**Comment by mfasold** [comment]
I found out that PGS000192 seems to lead to problems. Without it, there are no error messages for now.

However, the process is now running for 5 days, and with a current memory usage of 190 Gb I am running into swap memory. Thus, I am afraid it will not finish in reasonable time. Is there a way to reduce resource requirements for the combination of score files?

**Comment by smlmbrt** [comment]
Which part is using 190 GB, I assume the matching? There's not really a way to get around this - you could match smaller subsets of files and re-combine the matches to make the final scoring file, but we haven't really tested it at this scale.

**Comment by smlmbrt** [comment]
Hmmm, that's written to be memory-light so we'll have to check on that (cc @nebfield). But you can run it on batches of scores and then concatenate the outputs as they'll have the same column order.

**Comment by smlmbrt** [comment]
You shouldn't have the memory problem in v0.5.3, it looks like you're using v1.

**Comment by mfasold** [comment]
Yes, I was using v1.0. Thanks for the advice, I am trying version v0.5.3 now. It seemed like a nice performance upgrade first, running to item 65/4488 very quick. But two hours in, at item 78/4488, it seems to slow down and memory is up at 120Gb again.

**Comment by mfasold** [comment]
> But you can run it on batches of scores and then concatenate the outputs as they'll have the same column order.

I tried it with batches of 70 scores. This works with all but one batch (PGS003364 to PGS003433) where it stalls forever.

**Comment by nebfield** [comment]
Fixed in the [new release](https://pypi.org/project/pgscatalog-utils/1.0.1/):

https://github.com/PGScatalog/pygscatalog/pull/15

Thanks for the bug report 🚀

**Comment by mfasold** [comment]
Is it already possible to use this release in pgsc_calc? AFAICS, it is currently using the docker image ghcr.io/pgscatalog/pgscatalog_utils:v0.4.3

**Comment by nebfield** [comment]
We're planning to integrate [`pygscatalog`](https://github.com/PGScatalog/pygscatalog/) with pgsc_calc [in the next release](https://github.com/PGScatalog/pgsc_calc/milestone/5) of the calculator. 

The release should be fairly soon (within a few weeks). The largest amount of work will be setting up some automatic correlation tests to make sure calculated scores remain stable across releases. We normally do slow manual testing 😅

**Comment by mfasold** [comment]
Ok, thank you for the info. I was trying to run pgsc_calc with all the scores (stress test?) and it failed after 6h in the COMBINE_SCOREFILES step. So I am hoping that it will be possible with those fixes included in the next release.

**Comment by mfasold** [comment]
Remember that I had combined scorefiles in batches of 70 before. I want to obtain a list with all locations where there are effect alleles across all studies, to do variant calling on those positions. I find that if I collect those positions from the batched-combined scorefiles, I get 17M positions. If I take them from the one combined scorefile from the new release, I obtain only 2.6 M positions. Is there any reasonable explanation for such a loss in variant locations after combining scores? Unfortunately, I deleted my old parallel processing results.
